controller/src/main/java/com/flipkart/varadhi/controller/SuppressionService.java

```diff
@@ -0,0 +1,11 @@
+package com.flipkart.varadhi.controller;
+
+import com.flipkart.varadhi.qos.entity.ClientLoadInfo;
+import com.flipkart.varadhi.qos.entity.SuppressionData;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface SuppressionService {
```

Comment:
lets call it something else :)

If you look at it carefully, it is just a batch version of RateLimiter interface only.

```java
RateLimiter {
    boolean isAllowed(long quota)
}
```

Here the context in which it is going to be used is the 
`topic` and the fact that `a call to it means 1 request`

Your SuppressionService<R, L> looks like
```java
R  addTrafficData(L loadInfo)
```

From this it looks like RateLimiter is just a specialization of SuppressionService<Boolean, Long>.
So, we might be able to call SuppressionService just RateLimiter<R, L>.

Is it a coincidence that SuppressionService and RateLimiter's method's shape are similar?

Reply:
This is a service that handles all the client load information, renamed to `DistributedRateLimiterImpl`